OpenRA.Game/Traits/ActivityUtils.cs

```diff
@@ -44,7 +44,7 @@ public static Activity RunActivity(Actor self, Activity act)
 				else
 					start = current;
 
-				if (prev == act)
+				if (act == prev || act == prev.Parent)
```

Comment:
I might be missing something, but I currently don't see the problem.  The main thing i'm suggesting is a global replacement of NextActivity -> NextOrChildActivity in the traits being migrated to `CompositeActivity`.  I'd *really* like to know when reading the code whether an activity is finishing for good vs potentially deferring to a child activity, without having to remember to check whether it is a normal or composite activity.